### ISSUE TITLE
retworks build fails "ERROR: Failed building wheel for retworkx"

### ISSUE THREAD

**Issue by JanLahmann** [JanLahmann/RasQberry#33]
There seems to be a regression that prevents from building (installing) retworkx, which is a prerequisite for qiskit-aer.

Error message includes:
```  
error[E0599]: no method named `into_pyarray` found for struct `ndarray::ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>` in the current scope
     --> src/lib.rs:965:12
      |
  965 |     Ok(mat.into_pyarray(py).into())
      |            ^^^^^^^^^^^^ method not found in `ndarray::ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>`
  
  error[E0599]: no method named `into_pyarray` found for struct `ndarray::ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>` in the current scope
      --> src/lib.rs:1033:12
       |
  1033 |     Ok(mat.into_pyarray(py).into())
       |            ^^^^^^^^^^^^ method not found in `ndarray::ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>`
  
  error[E0599]: no method named `into_pyarray` found for struct `ndarray::ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>` in the current scope
      --> src/lib.rs:1210:15
       |
  1210 |     Ok(matrix.into_pyarray(py).into())
       |               ^^^^^^^^^^^^ method not found in `ndarray::ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>`
  
  error[E0599]: no method named `into_pyarray` found for struct `ndarray::ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>` in the current scope
      --> src/lib.rs:1281:15
       |
  1281 |     Ok(matrix.into_pyarray(py).into())
       |               ^^^^^^^^^^^^ method not found in `ndarray::ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>`
  
  error[E0599]: no method named `into_pyarray` found for struct `ndarray::ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>` in the current scope
      --> src/lib.rs:1325:15
       |
  1325 |     Ok(matrix.into_pyarray(py).into())
       |               ^^^^^^^^^^^^ method not found in `ndarray::ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>`
  
  error[E0599]: no method named `into_pyarray` found for struct `ndarray::ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>` in the current scope
      --> src/lib.rs:1369:15
       |
  1369 |     Ok(matrix.into_pyarray(py).into())
       |               ^^^^^^^^^^^^ method not found in `ndarray::ArrayBase<ndarray::OwnedRepr<f64>, ndarray::Dim<[usize; 2]>>`
  
  warning: unused import: `numpy::IntoPyArray`
    --> src/lib.rs:46:5
     |
  46 | use numpy::IntoPyArray;
     |     ^^^^^^^^^^^^^^^^^^
     |
     = note: `#[warn(unused_imports)]` on by default
  
  error: aborting due to 6 previous errors; 1 warning emitted
  
  For more information about this error, try `rustc --explain E0599`.
  error: could not compile `retworkx`.
  
  Caused by:
    process didn't exit successfully: `rustc --crate-name retworkx --edition=2018 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type cdylib --emit=dep-info,link -C opt-level=3 -Cembed-bitcode=no --crate-type cdylib -C metadata=1783fb939be7d2a2 --out-dir /tmp/pip-install-n4j5bmkj/retworkx/target/release/deps -L dependency=/tmp/pip-install-n4j5bmkj/retworkx/target/release/deps --extern fixedbitset=/tmp/pip-install-n4j5bmkj/retworkx/target/release/deps/libfixedbitset-8edc2f44888dd404.rlib --extern hashbrown=/tmp/pip-install-n4j5bmkj/retworkx/target/release/deps/libhashbrown-e871ca93b6602654.rlib --extern ndarray=/tmp/pip-install-n4j5bmkj/retworkx/target/release/deps/libndarray-0279521f60f59892.rlib --extern numpy=/tmp/pip-install-n4j5bmkj/retworkx/target/release/deps/libnumpy-036b558da4386307.rlib --extern petgraph=/tmp/pip-install-n4j5bmkj/retworkx/target/release/deps/libpetgraph-341750ba0c549f5f.rlib --extern pyo3=/tmp/pip-install-n4j5bmkj/retworkx/target/release/deps/libpyo3-433b4e8f96059ecb.rlib --extern rand=/tmp/pip-install-n4j5bmkj/retworkx/target/release/deps/librand-25ede033d8300093.rlib --extern rand_pcg=/tmp/pip-install-n4j5bmkj/retworkx/target/release/deps/librand_pcg-2d514a663aaff7b3.rlib --extern rayon=/tmp/pip-install-n4j5bmkj/retworkx/target/release/deps/librayon-ee7c0f42709cf910.rlib` (exit code: 1)
  cargo rustc --lib --manifest-path Cargo.toml --features pyo3/extension-module --release --verbose -- --crate-type cdylib
  error: cargo failed with code: 101
  
  ----------------------------------------
  ERROR: Failed building wheel for retworkx
Failed to build retworkx
ERROR: Could not build wheels for retworkx which use PEP 517 and cannot be installed directly
```

**Comment by JanLahmann** [comment]
As discussed in https://github.com/piwheels/piwheels/pull/263, there will be a new version of retworkx released soon.
This will probably meet all requirements so that a precompiled wheel file can be provided by pinwheels.org.
In the meantime, the following workaround can be used to install the latest version of Qiskit (0.23):

1. Start Qiskit 0.23 install using "rasqberry-config".
2. After the build of retworkx fails, activate the "rasqberry" venv, and do a "Building from source" as described in https://github.com/Qiskit/retworkx#building-from-source
```
$ rasqberry
$ git clone https://github.com/Qiskit/retworkx.git
$ cd retworkx/
$ pip install .
```
3. Complete the installation of Qiskit with
`$ pip install --prefer-binary 'qiskit[visualization]==0.23.*'`

**Comment by mtreinish** [comment]
You can actually shortcut the instructions a bit here, and replace the separate git clone and pip install into a single command with `pip install git+https://github.com/Qiskit/retworkx` which will have pip handle the git checkout and install as one step

**Comment by mtreinish** [comment]
Just a heads up I pushed out the retworkx 0.7.0 release a little while ago which should fix the issues installing from sdist. So the script should work again now.

**Comment by JanLahmann** [comment]
closed with #34 
Thank you @mtreinish !
